docs/faq.md

```diff
@@ -0,0 +1,422 @@
+<!--
+- Copyright (c) 2019, Arm Limited and Contributors
+-
+- SPDX-License-Identifier: Apache-2.0
+-
+- Licensed under the Apache License, Version 2.0 the "License";
+- you may not use this file except in compliance with the License.
+- You may obtain a copy of the License at
+-
+-     http://www.apache.org/licenses/LICENSE-2.0
+-
+- Unless required by applicable law or agreed to in writing, software
+- distributed under the License is distributed on an "AS IS" BASIS,
+- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+- See the License for the specific language governing permissions and
+- limitations under the License.
+-
+-->
+
+# Vulkan FAQ
```

Comment:
So my main concern with this file is it suppose to be a best practice mixed with some random things related to C++ and Android mixed in. I guess where are these questions coming from and do we really want a single large monolithic file for all of these. Not sure this is the best spot as the ./docs folder has been more related towards using the Vulkan Samples themselves and this seems like overloading the purpose of this directory

Like I imagine all these should be FAQ related to the samples or frameworks and not try to mix it with general best practices as it gets blurry what belongs here then

Reply:
I will let @AttilioProvenzano-ARM answer to this

Reply:
I think it makes sense to look for a different place for this, it made sense in the context of the [Mobile Best Practice](https://github.com/ARM-software/vulkan_best_practice_for_mobile_developers) project, but here it's not really related to samples and we don't want to create confusion.

I'd be happy to remove it from here and then rethink where this should belong.

Reply:
yes, but I don't want this work of this page to go to waste, its some good stuff, but lets move to a separate PR as I know some other Vulkan WG members who would want to take a closer look at this FAQ